internal/utils/endorsement.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/cauthdsl"
+	"github.com/hyperledger/fabric/msp"
+	"github.com/hyperledger/fabric/msp/mgmt"
+)
+
+func NewPolicyEvaluator() *PolicyEvaluator {
+	manager := mgmt.NewDeserializersManager(factory.GetDefault())
+	return &PolicyEvaluator{
+		IdentityDeserializer: manager.GetLocalDeserializer(),
+	}
+}
+
+type PolicyEvaluator struct {
+	msp.IdentityDeserializer
+}
+
+func (id *PolicyEvaluator) EvaluateIdentity(policyBytes []byte, identityBytes []byte) error {
+	identity, err := id.IdentityDeserializer.DeserializeIdentity(identityBytes)
+	if err != nil {
+		return fmt.Errorf("error while deserialzing identity, err: %s", err)
+	}
+
+	pp := cauthdsl.NewPolicyProvider(id.IdentityDeserializer)
+	policy, _, err := pp.NewPolicy(policyBytes)
+	if err != nil {
+		return err
+	}
+
+	return policy.EvaluateIdentities([]msp.Identity{identity})
+}
+
+func IsValidEndorserIdentity(identityBytes []byte, ccDef *lifecycle.QueryApprovedChaincodeDefinitionResult) error {
```

Comment:
This function is defined but only used in tests. Anyway, it should be helpful once the enclave's identity/key is handled through MSP.

Reply:
true, but before we introduce msp for the enclave, we can use this method to check whether a peer that hosts a chaincode enclave is valid endorsing peer (this only makes sense once we have some form of org-enclave binding).

Reply:
it is required to implement `can_endorse`?